internal/api/observability/secrets.go

```diff
@@ -11,3 +17,83 @@ func NewSecretReference(keyName, secretName string) *obs.SecretReference {
 		SecretName: secretName,
 	}
 }
+
+// Secrets is a map of secrets
+type Secrets map[string]*corev1.Secret
+
+// HashMD5 returns an MD5 representation of the secrets
+func (s Secrets) HashMD5() string {
```

Comment:
I would recommend not using MD5 as a hashing algorithm. Either use a "proper cryptographic hash" like SHA256/512 or use a newer non-cryptographic hash like [fnv](https://pkg.go.dev/hash/fnv) to signal "this has no security implications".